src/renderer/modules/webpack.ts

```diff
@@ -137,193 +144,149 @@ function getExports(m: RawModule): ModuleExports | undefined {
-function byPropsInternal<P extends string>(
-  props: P[],
-  all = false,
-): (ModuleExportsWithProps<P> | undefined) | Array<ModuleExportsWithProps<P>> {
-  const result = [];
-  for (const m of getAllRawModules()) {
-    const exp = getExportsForProps(m, props);
-    if (exp) {
-      if (all) {
-        result.push(exp);
-      } else {
-        return exp;
-      }
-    }
-  }
-  if (all) {
-    return result;
+  if (raw) {
+    return rawModule;
   }
-}
 
-export function byPropsFilter<P extends string>(props: P[]) {
-  return (m: RawModule): m is RawModuleWithProps<P> => Boolean(getExportsForProps(m, props));
+  return typeof rawModule !== "undefined" ? getExports(rawModule) : void 0;
 }
 
-export function getByProps<P extends string>(...props: P[]): ModuleExportsWithProps<P> | undefined {
-  return byPropsInternal(props, false);
-}
+// Searcher
+
+// I'd prefer to use conditional types instead of overloading here, but I had some weird issues with it
+export function getModule(
+  filter: Filter,
+  options?: { all?: false; raw?: false },
+): ModuleExports | undefined;
+export function getModule(filter: Filter, options?: { all?: true; raw?: false }): ModuleExports[];
+export function getModule(
+  filter: Filter,
+  options?: { all?: false; raw?: true },
+): RawModule | undefined;
+export function getModule(filter: Filter, options?: { all?: true; raw?: true }): RawModule[];
+
+export function getModule(
+  filter: Filter,
+  options: GetModuleOptions | undefined = {
+    all: false,
+    raw: false,
+  },
+): ModuleExports[] | RawModule[] | ModuleExports | RawModule | undefined {
+  if (typeof wpRequire?.c === "undefined") return options.all ? [] : void 0;
 
-export function getRawByProps<P extends string>(...props: P[]): RawModuleWithProps<P> | undefined {
-  return getRawModule(byPropsFilter(props)) as RawModuleWithProps<P> | undefined;
-}
+  const modules = options.all
+    ? Object.values(wpRequire.c).filter(filter)
+    : Object.values(wpRequire.c).find(filter);
 
-export function getRawById(id: number): RawModule | undefined {
-  if (!(id in wpRequire.c)) {
-    wpRequire(id);
+  if (options.raw) {
+    return modules;
   }
 
-  return wpRequire.c[id];
-}
-
-export function getById(id: number): ModuleExports | undefined {
-  const raw = getRawById(id);
-  if (typeof raw !== "undefined") {
-    return getExports(raw);
+  if (Array.isArray(modules)) {
+    return modules.map(getExports).filter((m): m is ModuleExports => typeof m !== "undefined");
+  } else if (modules) {
+    return getExports(modules);
   }
 }
 
-export function getAllByProps<P extends string>(...props: P[]): Array<ModuleExportsWithProps<P>> {
-  return byPropsInternal(props, true);
-}
+// Filters
 
-export function getAllRawByProps<P extends string>(...props: P[]): Array<RawModuleWithProps<P>> {
-  return getAllRawModules(byPropsFilter(props)) as Array<RawModuleWithProps<P>>;
-}
+export const filters = {
+  props<P extends string>(props: P[]) {
```

Comment:
I'd prefer calling this `byProps`

Reply:
I'm unsure on this one (@FifiTheBulldog ?)

Reply:
I second @asportnoy's opinion here.